src/faebryk/library/NetTie.py

```diff
@@ -0,0 +1,51 @@
+# This file is part of the faebryk project
+# SPDX-License-Identifier: MIT
+
+import logging
+from enum import Enum
+
+import faebryk.library._F as F
+from faebryk.core.module import Module
+from faebryk.core.moduleinterface import ModuleInterface
+from faebryk.library.Electrical import Electrical
+from faebryk.library.KicadFootprint import KicadFootprint
+from faebryk.libs.picker.picker import has_part_picked_remove
+
+logger = logging.getLogger(__name__)
+
+
+class NetTie[T: ModuleInterface](Module):
+    class Size(float, Enum):
+        _2_0MM = 2.0
+        _0_5MM = 0.5
+
+    unnamed: list[T]
+
+    def __init__(
+        self,
+        width: Size,
```

Comment:
Shouldn't the width be a parameter instead of constructor argument?

Reply:
Ideally, but then I'd need to defer construction until later to make the footprint trait work.
Until that's all working this seems simpler and clearer

Reply:
You don't need delayed construction, because you are not constructing anything. Adding a trait is always possible.
You can use `is_implemented` in the trait to check whether the width is 'ready' for making a footprint.

Reply:
```python

width: TBD[Quantity]

...

@L.rt_field
def footprint(self):
   class _(F.has_footprint):
      def get_footprint(_self):
          return F.KicadFootprint(f"NetTie:NetTie-2_SMD_Pad{width_mm:.1f}mm", pin_names=["1", "2"])

      def is_implemented(_self):
          return isinstance(self.width, Constant)

```